tests/data/test_order.py

```diff
@@ -0,0 +1,172 @@
+                     for _ in range(0, SIMULATOR_LAST_PRICES_TO_CHECK)]
+
+    # append validating trade
+    recent_trades.append({"price": min_price, "timestamp": time.time()})
+    assert order_inst.check_last_prices(recent_trades, max_price, inferior=True)
+
+    # test inferior FALSE
+    max_price = 10.454677
+    min_price = 2.4273
+    order_inst.creation_time = time.time()
+    recent_trades = [{"price": random_price(min_value=min_price, max_value=max_price), "timestamp": time.time()}
+                     for _ in range(0, SIMULATOR_LAST_PRICES_TO_CHECK)]
+
+    # append validating trade
+    recent_trades.append({"price": max_price, "timestamp": time.time()})
+    assert order_inst.check_last_prices(recent_trades,
+                                        random_price(min_value=min_price, max_value=max_price - 1),
+                                        inferior=False)
+
+    # test price not in last trades
+    # test inferior TRUE
+    max_price = 7456.15555632315
+    min_price = 1421.1488845
+    order_inst.creation_time = time.time()
+    recent_trades = [{"price": random_price(min_value=min_price, max_value=max_price), "timestamp": time.time()}
+                     for _ in range(0, SIMULATOR_LAST_PRICES_TO_CHECK)]
+
+    assert not order_inst.check_last_prices(recent_trades, min_price, inferior=True)
+
+    # test inferior FALSE
+    max_price = 0.0001243753
+    min_price = 0.000012557753
+    order_inst.creation_time = time.time()
+    recent_trades = [{"price": random_price(min_value=min_price, max_value=max_price), "timestamp": time.time()}
+                     for _ in range(0, SIMULATOR_LAST_PRICES_TO_CHECK)]
+
+    assert not order_inst.check_last_prices(recent_trades,
+                                            max_price,
+                                            inferior=False)
+
+
+async def test_new(trader):
+    config, exchange_manager_inst, trader_inst = trader
+
+    # with real trader
+    order_inst = Order(trader_inst)
+    order_inst.update(order_type=TraderOrderType.BUY_MARKET,
+                      symbol="BTC/USDT",
+                      current_price=10000,
+                      quantity=1)
+
+    assert order_inst.order_type == TraderOrderType.BUY_MARKET
+    assert order_inst.symbol == "BTC/USDT"
+    assert order_inst.created_last_price == 10000
+    assert order_inst.origin_quantity == 1
+    assert order_inst.creation_time != 0
+    assert order_inst.get_currency_and_market() == ('BTC', 'USDT')
+    assert order_inst.side is None
+    assert order_inst.status == OrderStatus.OPEN
+
+    order_inst.update(order_type=TraderOrderType.STOP_LOSS_LIMIT,
+                      symbol="ETH/BTC",
+                      quantity=0.1,
+                      quantity_filled=5.2,
+                      price=0.12,
+                      stop_price=0.9)
+    assert order_inst.origin_stop_price == 0.9
+    assert order_inst.origin_price == 0.12
```

Comment:
should we also tests other computed fields ? like filled_quantity, exchange_order_type, filled_price and executed_time

Reply:
You want to test default values or set other values to update ?

Reply:
The ones I listed are computed values during ```update```, I don't think default values are worth testing but computed ones we should test